public/app/features/explore/ExploreTimeControls.tsx

```diff
@@ -44,6 +45,12 @@ export class ExploreTimeControls extends Component<Props> {
       from: adjustedFrom,
       to: adjustedTo,
     });
+
+    if (typeof adjustedFrom === 'string' && adjustedFrom.startsWith('now-')) {
```

Comment:
> if the user changes the range manually like below it will be tracked with the same event
I know and that's not what we intend to measure. The reason why I implemented it anyway is that it is the closest we can get to what we are actually interested in. With tracking all range selections I am a bit worried about the amount of data we would get and don't use. So what about tracking only the time ranges with timeRange.raw.to = "now"?

Reply:
Not sure if we need to be worried about the amount of data. In my opinion, if we can't get the correct data it's better to track something else (but correctly) or not to track. Having tracking data that may not represent what we're supposed to track may cause making incorrect decisions based on that data.

However, if we need to get "close enough" to quick time ranges and track it this way I would make sure it's clear what we're tracking:
* `grafana_explore_time_picker_time_range` -> we need to change the name of the event because it's not tracking all range selections. It could be for example `grafana_explore_time_picker_relative_to_time_range_selected`
* we need to make a comment in the code about why we're tracking relative ranges only (i.e. because we want to find out what quick choices are most popular).

Reply:
I agree that tracking the wrong things could lead to incorrect conclusions. If we don't have an issue with too much data let's track all time ranges. So we are going to have all the data and we can choose what we are interested in afterwards (e.g. in the dashboard). 
I will adjust the tracking there.